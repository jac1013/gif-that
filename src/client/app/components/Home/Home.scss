@import '../../../main.scss';

.home-section {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 960px;

  &__text {
    justify-content: center;
    align-items: center;
    font-size: 100px;
    font-weight: 500;
    line-height: 1.3;
    color: #fff;
    margin: 0 30px 0 30px;
  }
}
