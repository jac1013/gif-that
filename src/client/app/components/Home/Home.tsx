import * as React from 'react';
import {Link} from 'react-router-dom';

export const Home = () => (
  <section className='home'>
    This is our Home, go to the <Link to='/counter'>Counter!</Link>
  </section>
);

