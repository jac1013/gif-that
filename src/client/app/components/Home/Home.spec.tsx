import {suite, test} from 'mocha-typescript';
import {expect} from 'chai';
import * as React from "react";
import * as renderer from 'react-test-renderer';
import {ReactTestRenderer} from 'react-test-renderer';
import {MemoryRouter} from 'react-router';
import {Home} from './Home';

@suite
class HomeTest {
  component: ReactTestRenderer;
  before() {
    this.component = renderer.create(
      <MemoryRouter><Home/></MemoryRouter>
    );
  }

  @test
  shouldRenderCorrectly() {
    expect(this.component.toJSON()).to.matchSnapshot();
  }
}
