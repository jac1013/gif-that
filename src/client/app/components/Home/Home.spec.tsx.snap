// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeTest shouldRenderCorrectly 1`] = `
<section
  className="home-section"
>
  <div
    className="home-section__text"
  >
     Welcome to 
    <a
      className="home-section__counter-link"
      href="/counter"
      onClick={[Function]}
    >
      Counter!
    </a>
  </div>
</section>
`;
